lib/domain/usecase/send_images_interactor.dart

```diff
@@ -0,0 +1,79 @@
+import 'dart:collection';
+
+import 'package:fluffychat/presentation/extensions/asset_entity_extension.dart';
+import 'package:fluffychat/presentation/extensions/room_extension.dart';
+import 'package:matrix/matrix.dart';
+import 'package:photo_manager/photo_manager.dart';
+
+typedef TransactionId = String;
+
+typedef FakeImageEvent = SyncUpdate;
+
+class SendImagesInteractor {
+  final imageCacheQueue = Queue();
```

Comment:
we have something relate to state of this instance, please inject it as a factory